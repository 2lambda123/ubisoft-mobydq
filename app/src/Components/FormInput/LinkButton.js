import React from 'react';
import Button from '@material-ui/core/Button';

import { Link } from 'react-router-dom';

const LinkButton = ({
  type,
  color,
  variant,
  disabled,
  label,
  touched, // eslint-disable-line
  to
}) => <Button type={type} component={Link} to={to} disabled={disabled} variant={variant} color={color}>
  {label}
</Button>;

export default LinkButton;
