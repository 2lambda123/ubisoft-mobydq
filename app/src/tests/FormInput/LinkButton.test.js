import React from 'react';
import { shallowWrap } from './../../setupTests';

import LinkButton from '../../Components/FormInput/LinkButton';

describe('LinkButton unit test', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowWrap(
      <LinkButton
        type="submit"
        disabled={false}
        label="Button"
        variant="contained"
        color="primary"
        to="/"
      />
    );
  });

  it('renders', () => {
    expect(wrapper).toHaveLength(1);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
