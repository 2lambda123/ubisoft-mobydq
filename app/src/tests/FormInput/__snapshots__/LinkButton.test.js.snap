// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton unit test matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LinkButton
    color="primary"
    disabled={false}
    label="Button"
    to="/"
    type="submit"
    variant="contained"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": "Button",
      "color": "primary",
      "component": [Function],
      "disabled": false,
      "to": "/",
      "type": "submit",
      "variant": "contained",
    },
    "ref": null,
    "rendered": "Button",
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": "Button",
        "color": "primary",
        "component": [Function],
        "disabled": false,
        "to": "/",
        "type": "submit",
        "variant": "contained",
      },
      "ref": null,
      "rendered": "Button",
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "childContextTypes": Object {
      "router": [Function],
    },
    "context": Object {
      "router": Object {
        "history": Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        },
        "route": Object {
          "location": Object {},
          "match": Object {},
        },
      },
    },
  },
}
`;
